--- cli/Makefile	2020-10-06 10:50:59.000000000 -0700
+++ Makefile	2020-10-06 10:50:55.000000000 -0700
@@ -25,10 +25,6 @@
 endif


-.PHONY: clean
-clean:
-	rm -rf $(shell pwd)/dist
-
 .PHONY: generate
 generate:
 	@go generate ./...
@@ -250,3 +246,12 @@
 doctoc:
 	npx doctoc README.md

+clean:
+
+distclean:
+
+%:
+	cp debian/Makefile Makefile.debian
+	$(MAKE) -f Makefile.debian $@
+	rm -f Makefile.debian
+
